### PR TITLE
Reorganize Swift sample code

### PR DESCRIPTION
**Pull request by gfontenot**
This is kind of a weird change, but this file is fairly unruly and I'm
not sure how best to structure it so that it's useful. I ended up
removing a couple of things that I didn't think were needed and changed
up the sections to be a bit more structured.

It's probably best to read this in a split diff so that you can read the new
version of the file as is. If anyone has any objection to the removal of
things, feel free to speak up.

ping @thoughtbot/ios

### CODE REVIEW

**Comment by sharplet** [comment]
Seems legit👍🏻

**Comment by gfontenot** [comment]
@thoughtbot/ios any comments here?

**Comment by tonyd256** [comment]
👍

**Comment by tonyd256** [comment]
is this ready to go? looks like people are cool with it if you want to merge.

Also, if you want to update it for swift 3 I'm ok with that here ... or wait for a separate PR is fine too.

**Comment by jamescmartinez** [comment]
:shipit:

File: style/swift/sample.swift [hunk 1]
```diff
@@ -1,24 +1,33 @@
...
-// MARK: Types
-
-// Prefer structs over classes
+// Types begin with a capital letter
 struct User {
   let name: String
+
+  // if the first letter of an acronym is lowercase, the entire thing should
+  // be lowercase
+  let json: AnyObject
```
Comment (sharplet) [inline style/swift/sample.swift]: `AnyObject` => `Any`

File: style/swift/sample.swift [hunk 2]
```diff
@@ -1,24 +1,33 @@
...
+
+  // if the first letter of an acronym is uppercase, the entire thing should
+  // be uppercase
+  static func decodeFromJSON(json: AnyObject) -> User {
```
Comment (sharplet) [inline style/swift/sample.swift]: `decode(from json: Any)`

File: style/swift/sample.swift [hunk 3]
```diff
@@ -1,24 +1,33 @@
...
 final class MyViewController: UIViewController {
   // Prefer strong IBOutlet references
   @IBOutlet var button: UIButton!
 }
 
-// MARK: Closures
-
 // Use typealias when closures are referenced in multiple places
-typealias CoolClosure = (foo: Int) -> Bool
+typealias CoolClosure = Int -> Bool
```
Comment (sharplet) [inline style/swift/sample.swift]: `(Int) -> Bool`

**Pull request merged**